components/icu4x_ecma402/src/pluralrules.rs

```diff
@@ -0,0 +1,340 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE )
+
+//! Implements the traits found in [ecma402_traits::pluralrules].
+
+use icu_plurals as ipr;
+use icu_plurals::PluralRulesError;
+
+pub(crate) mod internal {
+    use ecma402_traits::pluralrules::options::Type;
+    use ecma402_traits::pluralrules::Options;
+    use icu_plurals::{PluralCategory, PluralOperands, PluralRuleType};
+    use std::cmp::{max, min};
+
+    // The difference between `first` and `second`, clamped on the bottom by zero.
+    fn clamp_diff(first: usize, second: usize) -> usize {
+        match first > second {
+            true => first - second,
+            false => 0,
+        }
+    }
+
+    /// Converts the trait style option to an equivalent ICU4X type.
+    pub fn to_icu4x_type(style: &Type) -> PluralRuleType {
+        match style {
+            Type::Ordinal => PluralRuleType::Ordinal,
+            Type::Cardinal => PluralRuleType::Cardinal,
+        }
+    }
```

Comment:
Can you do `Into<PluralRuleType> for Type` here?

Reply:
I thought you needed to own at least one of the types you're implementing `Into` for.

Reply:
Ah, sorry. I was confused by where `Type` lives. You're right. Disregard my comment pls.